client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -4,6 +4,20 @@ import { useSelector } from 'calypso/state';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import { getSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { ExperimentAssignment } from '@automattic/explat-client';
+import type { ExperimentOptions } from '@automattic/explat-client-react-helpers';
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+let useExperiment = (
+	experimentName: string,
+	options?: ExperimentOptions
+): [ boolean, ExperimentAssignment | null ] => [ false, null ];
+/* eslint-enable @typescript-eslint/no-unused-vars */
+try {
+	if ( typeof window !== undefined ) {
+		( { useExperiment } = await import( 'calypso/lib/explat' ) );
+	}
+} catch ( e ) {}
```

Comment:
Hey @mmtr thanks for addressing this. As for the build failure I think calypso might not fully support [top level await yet](https://github.com/tc39/proposal-top-level-await). Reverting to @tyxla for possible feedback on this.

To be specific the section loader seems to be failing.
<img width="941" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3422709/d2e1331d-e533-40dc-8256-ef69ea16f414">

---

In the meantime can you probably try this with an asyc IIFE? @jessie-ross hoping this approach should be sufficient 🤔 .

```typescript
( async function () {
	try ...
        ...
           ... await 
        ...
} )();
```


Also, note that the `typeof` operator always returns the string so the comparison should be `typeof window !== undefined`

Reply:
Regarding the top-level await, it's not enabled yet, but totally feel free to enable it in a separate PR (I'm happy to help with review) and experiment with it. 

FWIW Webpack offers support for it and it can be enabled with the [`topLevelAwait` experiment](https://webpack.js.org/configuration/experiments/#experimentstoplevelawait). It will be automatically updated in newer versions of Webpack. 

Alternatively, an async IIFE sounds good if you're looking for a quick workaround 👍

Reply:
Thanks Dehan! That fixed the build error 🙂 . I'm now facing another error when visiting the logged-out showcase which produces a blank screen:

```
react-dom.development.js:16572 Uncaught Error: Should have a queue. This is likely a bug in React. Please file an issue.
    at updateReducer (react-dom.development.js:16572:11)
    at updateState (react-dom.development.js:17004:10)
    at Object.useState (react-dom.development.js:17915:16)
    at useState (react.development.js:1623:21)
    at useExperiment (index.js:26:85)
    at useSiteGlobalStylesStatus (use-site-global-styles-status.ts:71:36)
    at eval (index.jsx:393:121)
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateFunctionComponent (react-dom.development.js:19583:20)
    at beginWork (react-dom.development.js:21596:16)
```
Not sure what's causing it. Any ideas?

Reply:
This happens because practically we're breaking the [rules of hooks](https://react.dev/warnings/invalid-hook-call-warning). On the one hand, it's because we're overriding the hook, and React won't appreciate that. On the other, it's because we're calling `useExperiment()` in a place that is not a functional component or hook.

React treats hooks in a very special, organized way, and that helps React optimize for their performance under the hood. That's the main reason why they're only allowed at the top level of a functional component or hook, and why they can't be called conditionally - a condition would reorder or reorganize the original, predictable order of hooks used in a certain hook or component.

I'd recommend not using hooks at all for the dynamic async part, since that doesn't seem to be properly handled by React. Instead, we could have a hook with an async wrapper like it's [shown here](https://dev.to/danialdezfouli/what-s-wrong-with-the-async-function-in-useeffect-4jne), and internally we wouldn't be calling a hook (`useExperiment()`) but rather just an async function that does the same.

Reply:
If this isn't running in a react context we should be using the async [`loadExperimentAssignment`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/explat-client/README.md#api-explatclientloadexperimentassignment).

Reply:
Thanks everyone once more again! I think I managed to get it to work 🎉